aws/s3_test.go

```diff
@@ -33,39 +33,67 @@ func TestMain(m *testing.M) {
 	os.Exit(exitVal)
 }
 
-// TestS3Bucket represents a test S3 bucket.
-type TestS3Bucket struct {
-	name        string
-	tags        []map[string]string
-	isVersioned bool
+// genTestsBucketName generates a test bucket name.
+func genTestBucketName() string {
+	return strings.ToLower("cloud-nuke-test-" + util.UniqueID() + util.UniqueID())
 }
 
-// SetupInfo has test case setup info.
-type SetupInfo struct {
+// createSession creates a new session and returns it.
+func createSession(region string) (*session.Session, error) {
+	if region == "" {
+		var err error
+		region, err = getRandomRegion()
+		if err != nil {
+			return nil, err
+		}
+		logging.Logger.Debugf("Creating session in region - %s", region)
+	}
+	session, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region)},
+	)
+	return session, err
+}
+
+// AWSParams has AWS params info.
+type AWSParams struct {
 	region     string
 	awsSession *session.Session
 	svc        *s3.S3
-	bucketName string
 }
```

Comment:
NIT: rename to something that is specific to S3.

Reply:
This has been removed so not applicable.